File: apps/windows-native/WordZ.Windows/Contracts/EngineContracts.cs
namespace WordZ.Windows.Contracts;

public static class EngineContracts
{
    public const string JsonRpcVersion = "2.0";

    public static class Methods
    {
        public const string AppGetInfo = "app.getInfo";
        public const string AppGetPendingLaunchFiles = "app.getPendingLaunchFiles";
        public const string AppConsumeCrashRecoveryState = "app.consumeCrashRecoveryState";
        public const string LibraryList = "library.list";
        public const string LibraryOpenQuickPath = "library.openQuickPath";
        public const string LibraryImportPaths = "library.importPaths";
        public const string LibraryOpenSaved = "library.openSaved";
        public const string LibraryOpenSavedBatch = "library.openSavedBatch";
        public const string LibraryRenameCorpus = "library.renameCorpus";
        public const string LibraryMoveCorpus = "library.moveCorpus";
        public const string LibraryDeleteCorpus = "library.deleteCorpus";
        public const string LibraryCreateFolder = "library.createFolder";
        public const string LibraryRenameFolder = "library.renameFolder";
        public const string LibraryDeleteFolder = "library.deleteFolder";
        public const string LibrarySearchKwic = "library.searchKwic";
        public const string WorkspaceGetState = "workspace.getState";
        public const string WorkspaceSaveState = "workspace.saveState";
        public const string AnalysisStartTask = "analysis.startTask";
        public const string AnalysisCancelTask = "analysis.cancelTask";
        public const string AnalysisGetTaskState = "analysis.getTaskState";
        public const string DiagnosticsGetState = "diagnostics.getState";
        public const string UpdateGetState = "update.getState";
        public const string EngineShutdown = "engine.shutdown";
    }
}

File: apps/macos-native/WordZMac/Sources/WordZMac/Models/EngineContracts.swift
import Foundation

enum EngineContracts {
    static let jsonRpcVersion = "2.0"

    enum Method {
        static let appGetInfo = "app.getInfo"
        static let libraryList = "library.list"
        static let libraryImportPaths = "library.importPaths"
        static let libraryOpenSaved = "library.openSaved"
        static let libraryRenameCorpus = "library.renameCorpus"
        static let libraryMoveCorpus = "library.moveCorpus"
        static let libraryDeleteCorpus = "library.deleteCorpus"
        static let libraryCreateFolder = "library.createFolder"
        static let libraryRenameFolder = "library.renameFolder"
        static let libraryDeleteFolder = "library.deleteFolder"
        static let libraryListRecycleBin = "library.listRecycleBin"
        static let libraryRestoreRecycleEntry = "library.restoreRecycleEntry"
        static let libraryPurgeRecycleEntry = "library.purgeRecycleEntry"
        static let libraryBackup = "library.backup"
        static let libraryRestore = "library.restore"
        static let libraryRepair = "library.repair"
        static let workspaceGetState = "workspace.getState"
        static let workspaceSaveState = "workspace.saveState"
        static let workspaceGetUiSettings = "workspace.getUiSettings"
        static let workspaceSaveUiSettings = "workspace.saveUiSettings"
        static let analysisStartTask = "analysis.startTask"
        static let engineShutdown = "engine.shutdown"
    }

    enum TaskType {
        static let stats = "stats"
        static let compare = "compare"
        static let chiSquare = "chi-square"
        static let ngram = "ngram"
        static let kwic = "kwic"
        static let collocate = "collocate"
        static let wordCloud = "word-cloud"
        static let locator = "locator"
    }

    enum Event {
        static let engineReady = "engine.ready"
        static let engineStartupError = "engine.startupError"
        static let taskCompleted = "task.completed"
        static let taskFailed = "task.failed"
        static let taskCancelled = "task.cancelled"
    }
}
